File: SpeciAssist/src/main/java/com/itma/speciassist/model/VideoMentor.java
package com.itma.speciassist.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Data;

@Data
@Entity
public class VideoMentor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String url;
    private String fileName;
    private String title;
    
    
    @ManyToOne
    @JoinColumn(name = "mentor_id", nullable = false)
    @JsonIgnore
    private Mentor mentor;

    @ManyToOne
    @JoinColumn(name = "carriere_id")
    private Carriere carriere;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

   
}

File: frontend/src/app/model/video-mentor.ts
import { Carriere } from './carriere.model';
import { Mentor } from './mentor.model';
import { User } from './user';

export interface VideoMentor {
  id: number;
  fileName: string;
  title: string;
  url: string;
  carriere: Carriere;
  mentor: Mentor;
  user: User
}

// Ajoutez ceci pour obtenir carriereId et mentorId
export class VideoMentorImpl implements VideoMentor {
  id: number;
  fileName: string;
  title: string;
  url: string;
  carriere: Carriere;
  mentor: Mentor;
  user: User

  constructor(id: number, fileName: string, title: string, url: string, carriere: Carriere, mentor: Mentor, user: User) {
    this.id = id;
    this.fileName = fileName;
    this.title = title;
    this.url = url;
    this.carriere = carriere;
    this.mentor = mentor;
    this.user = user;
  }

  get carriereId(): number {
    return this.carriere.id;
  }

  get mentorId(): number {
    return this.mentor.id;
  }
}
